### ISSUE TITLE
Wrong height of Grid inside TabSheet

### ISSUE THREAD

**Issue by onlineandy** [vaadin/framework#11087]
**Vaadin Framework version**
8.4.5

**Browser version**

- Chrome 67
- IE 10
- (but does not really matter)

**Web container name and version**
Jetty 9.3.9

**Description of the bug**

The height of the Grid on the first Tab is miscalculated when switching between the tabs. It resizes correctly when pressing the button.
This seems to be a regression introduced in 8.4.2. In 8.4.1 the behavior is better, although not perfect.

**Minimal reproducible example**

```
final VerticalLayout layout = new VerticalLayout();

List<String> data = new ArrayList<>();
data.add("Data");
data.add("Data");
data.add("Data");
data.add("Data");
data.add("Data");
data.add("Data");
data.add("Data");
data.add("Data");
data.add("Data");
data.add("Data");

Panel panel = new Panel("Panel Tab");
panel.setHeight(100, Unit.PERCENTAGE);

TabSheet tabSheet = new TabSheet();
tabSheet.addTab(panel);
Label l = new Label("Just a label");
l.setCaption("Tab 2");
tabSheet.addTab(l);

Button button = new Button("Add Data");

Grid<String> grid = new Grid<>();
grid.addColumn(String::toString).setCaption("Test");
ListDataProvider<String> provider = DataProvider.ofCollection(data);
grid.setDataProvider(provider);

grid.setHeightMode(HeightMode.UNDEFINED);
grid.setRowHeight(50);

button.addClickListener(new ClickListener() {

	@Override
	public void buttonClick(ClickEvent event) {
		data.add("Data");
		grid.getDataProvider().refreshAll();
	}
});

VerticalLayout vl = new VerticalLayout();
vl.addComponent(grid);

panel.setContent(vl);
layout.addComponents(button, tabSheet);
```

**Expected behavior**

The Grid is fully visible when switching back to the first tab.

**Actual behavior**

The Grid is visually cut off at the bottom (see screenshot). It resizes correcty when pressing the button.

![2018-07-26-vaadin-bug](https://user-images.githubusercontent.com/851839/43248033-db8e7e36-90b6-11e8-95da-cdd150bbe0bc.png)

**Comment by mehdi-vaadin** [comment]
It's not related to `TabSheet` or `Panel`. `Grid` on its own has this bug.

**Comment by onlineandy** [comment]
Is there already a snapshot I can test with?

And when will there be a release with this fix? 8.4 oder only 8.5?

**Comment by mehdi-vaadin** [comment]
It will be fixed in 8.5.1.
